engines/python/setup/djl_python/huggingface.py

```diff
@@ -498,39 +533,81 @@ def register_adapter(inputs: Input):
     """
     adapter_name = inputs.get_property("name")
     adapter_path = inputs.get_property("src")
+    adapter_pin = inputs.get_property("pin") == "true"
     if not os.path.exists(adapter_path):
         raise ValueError(
             f"Only local LoRA models are supported. {adapter_path} is not a valid path"
         )
-    logging.info(f"Registering adapter {adapter_name} from {adapter_path}")
     _service.adapter_registry[adapter_name] = inputs
-    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
-        if isinstance(_service.model, PeftModel):
-            _service.model.load_adapter(adapter_path, adapter_name)
-        else:
-            _service.model = PeftModel.from_pretrained(_service.model,
-                                                       adapter_path,
-                                                       adapter_name)
 
-        if isinstance(_service.hf_pipeline, Pipeline):
-            _service.hf_pipeline.model = _service.model
+    try:
+        result = _service.add_lora(adapter_name, adapter_path)
+        if not result:
+            return Output().error(
+                "error",
+                message=f"Failed to register LoRA adapter {adapter_name}")
+
+        if adapter_pin:
+            result = _service.pin_lora(adapter_name)
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to pin LoRA adapter {adapter_name}")
+    except (ValueError, RuntimeError) as e:
```

Comment:
For the Memory errors, is there any chance we have access to specific error types in the backend rather than relying on parsing the strings/cause from the exception to determine whether it is a Memory error or not?

Reply:
I hope there's better way. Right now the error is thrown as:

https://github.com/vllm-project/vllm/blob/main/vllm/lora/models.py#L375-L376

https://github.com/vllm-project/vllm/blob/main/vllm/lora/worker_manager.py#L190-L193

I agree this way of catching is error prone if vllm change the error message. But I can't find other way.